src/Confluent.Kafka/Internal/Util.cs

```diff
@@ -33,14 +34,41 @@ public static string PtrToStringUTF8(IntPtr strPtr)
                 var length = 0;
                 unsafe
                 {
-                    byte* pTraverse = (byte *)strPtr;
+                    byte* pTraverse = (byte*)strPtr;
                     while (*pTraverse != 0) { pTraverse += 1; }
                     length = (int)(pTraverse - (byte*)strPtr);
                 }
                 var strBuffer = new byte[length];
                 System.Runtime.InteropServices.Marshal.Copy(strPtr, strBuffer, 0, length);
                 return Encoding.UTF8.GetString(strBuffer);
             }
+
+            public static T PtrToStructure<T>(IntPtr ptr)
+            {
+#if NET45
+                return (T)SystemMarshal.PtrToStructure(ptr, typeof(T));
+#else
```

Comment:
The only potential concern I have with this PR is that this would be less efficient than the generic version and it's used every time `ConsumerPoll` handles a new message, which is frequently. I think it should be a very minimal hit though - `typeof(T)` doesn't allocate a new `Type` class on every invocation, so the GC isn't being pressured. Also, `Message` is a class, so there is no unboxing going on.